infra/staging/main.tf

```diff
@@ -151,6 +151,66 @@ resource "google_storage_bucket_iam_member" "frontend_bucket_viewer" {
   role   = "roles/storage.objectViewer"
   member = "allUsers"
 }
+#######################################
+## Load balancer for frontend bucket ##
+#######################################
+
+// Following the guide here: https://cloud.google.com/cdn/docs/setting-up-cdn-with-bucket
+
+// Enable Compute Engine API to setup an IP address
+resource "google_project_service" "compute_engine_api" {
+  project = data.google_project.project.project_id
+  service = "compute.googleapis.com"
+}
+
+// Setup IP address for load balancer
+resource "google_compute_global_address" "call_home_frontend_ip" {
+  project = data.google_project.project.project_id
+  name = "call-home-frontend-ip"
+}
+
+// Backend bucket with CDN policy with default ttl settings
+resource "google_compute_backend_bucket" "call_home_frontend_lb" {
+  name = "call-home-frontend-lb"
+  description = "Load balancer for the frontend GCS bucket"
+  project = data.google_project.project.project_id
+  bucket_name = google_storage_bucket.call_home_frontend.name
+  enable_cdn = true
+
+  cdn_policy {
+    cache_mode = "CACHE_ALL_STATIC"
+    client_ttl = 3600
+    default_ttl = 3600
+    max_ttl = 86400
+    negative_caching = true
+    serve_while_stale = 86400
+  }
+}
+
+// URL map for the backend bucket
+resource "google_compute_url_map" "call_home_frontend_lb_url_map" {
+  name = "call-home-frontend-lb-url-map"
+  project = data.google_project.project.project_id
+  default_service = google_compute_backend_bucket.call_home_frontend_lb.id
+}
```

Comment:
@halfwhole @chaitanyabaranwal I suspect the url map is how we'll end up connecting Strapi to the load balancer. It seems likely that we'll add some rules to the url map (https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/compute_url_map#example-usage---url-map-bucket-and-service) to direct different urls to different backends - either the GCS bucket or the App Engine app.

e.g.I think we might want something like

- Direct `app2[-staging].callhome.sg` to the GCS bucket
- Direct `app2[-staging]-api.callhome.sg` or `app2[-staging].callhome.sg/api` to the Strapi App Engine app.

which _maybe_ translates into something like 

```
  host_rule {
    hosts        = ["app2-staging.callhome.sg"]
    path_matcher = "callhome"
  }
  path_matcher {
    name            = "callhome"
    default_service = my_bucket.static.id

    path_rule {
      paths   = ["/api"]
      service = my_app_engine.id
    }
  }
```